#!/usr/bin/env python
PACKAGE = 'jaguar'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
gen.add('gain_p', double_t, 1, 'Proportional Gain', 0, -10, 10)
gen.add('gain_i', double_t, 2, 'Integral Gain', 0, -10, 10)
gen.add('gain_d', double_t, 4, 'Differential Gain', 0, -10, 10)
gen.add('brake', bool_t, 8, 'Brake the motors when stopped', False)
gen.add('cpr', int_t, 16, 'Encoder counts per wheel revolution.', 1, 1, 65535)
gen.add('wheel_diameter', double_t, 32, 'Diameter of the wheels, in meters.', 0.254)
gen.add('wheel_separation', double_t, 64, 'Distance between the wheels, in meters.', 0.381)
gen.add('odom_rate', int_t, 128, 'Odometry update rate in milliseconds.', 50, 1, 255)
gen.add('diag_rate', int_t, 256, 'Diagnostics update rate in milliseconds.', 200, 1, 255)
gen.add('heartbeat_rate', int_t, 512, 'Heartbeat rate', 100, 1, 255)
gen.add('alpha', double_t, 2048, 'Noise ratio', 0.0, 0.0, 1.0)

# for testing purposes only
gen.add('setpoint', double_t, 1024, 'Velocity Setpoint', 0, -300, +300)

exit(gen.generate(PACKAGE, 'jaguar', 'Jaguar'))
# vim: set ft=python:
